src/components/TransactionBlock/WalletTransactionBlock.tsx

```diff
@@ -0,0 +1,737 @@
+import { Chain, supportedChains } from '../../utils/chain';
+import NetworkAssetSelectInput from '../NetworkAssetSelectInput';
+import { IAssetWithBalance } from '../../providers/EtherspotContextProvider';
+import { CombinedRoundedImages, RoundedImage } from '../Image';
+import { Pill } from '../Text';
+import { Theme } from '../../utils/theme';
+import AccountSwitchInput from '../AccountSwitchInput';
+import { IMultiCallData, ITransactionBlock } from '../../types/transactionBlock';
+import {
+  WalletAssetSearchIcon,
+  WalletBridgeIcon,
+  WalletDepositIcon,
+  WalletSendIcon,
+  WalletSwapIcon,
+  WalletWithdrawIcon,
+  WalletDropdownUpIcon,
+  WalletDropdownDownIcon,
+  WalletCopyIcon,
+  WalletCloseSearchIcon,
+} from './Icons';
+import { Text } from '../Text';
+import { TRANSACTION_BLOCK_TYPE, TRANSACTION_BLOCK_TYPE_KEY } from '../../constants/transactionBuilderConstants';
+import { ISendAssetTransactionBlockValues } from './SendAssetTransactionBlock';
+
+interface IChainAssets {
+  title: string;
+  chainId: number;
+  chain: Chain;
+  assets: IAssetWithBalance[];
+}
+
+export interface IWalletTransactionBlock {
+  chain?: Chain;
+  availableTransactionBlocks: ITransactionBlock[];
+  hasTransactionBlockAdded: boolean;
+  addTransactionBlock: (block: ITransactionBlock, isBridgeDisabled: boolean, isKlimaIncluded: boolean) => void;
+}
+
+const WalletTransactionBlock = ({
+  chain,
+  availableTransactionBlocks,
+  hasTransactionBlockAdded,
+  addTransactionBlock,
+}: IWalletTransactionBlock) => {
+  const theme: Theme = useTheme();
+
+  const { setTransactionBlockValues, resetTransactionBlockFieldValidationError } = useTransactionBuilder();
+
+  const {
+    providerAddress,
+    accountAddress,
+    getSupportedAssetsWithBalancesForChainId,
+    smartWalletOnly,
+    smartWalletBalanceByChain,
+    sdk,
+  } = useEtherspot();
+
+  const [selectedNetwork, setSelectedNetwork] = useState<Chain | null>(supportedChains[1]);
+  const [fetchingAssets, setFetchingAssets] = useState(false);
+  const [showAllChains, setShowAllChains] = useState(true);
+
+  const [showChainDropdown, setShowChainDropdown] = useState(false);
+  const [selectedChains, setSelectedChains] = useState<number[]>([supportedChains[1].chainId]);
+  const [hideChainList, setHideChainList] = useState<number[]>([]);
+  const [displayAssets, setDisplayAssets] = useState<IAssetWithBalance[]>([]);
+
+  const [walletTotal, setWalletTotal] = useState(0);
+  const [chainAssets, setChainAssets] = useState<IChainAssets[] | null>(null);
+
+  const [showSearch, setShowSearch] = useState(false);
+  const [searchValue, setSearchValue] = useState('');
+
+  useEffect(() => {
+    const getAssets = async () => {
+      if (!chainAssets || fetchingAssets) {
+        setFetchingAssets(true);
+        let allAssets: IChainAssets[] = [];
+        supportedChains.map(async (chain, i) => {
+          try {
+            if (chain.chainId === 43114) return;
```

Comment:
Avalance? please add a comment why and name of chain for ID, thanks!

Reply:
done, basically just triggers another sign modal when the block loads so i removed it - I've added a comment now